src/main/clojure/clara/rules/engine.cljc

```diff
@@ -29,12 +29,12 @@
 ;; Token with no bindings, used as the root of beta nodes.
 (def empty-token (->Token [] {}))
 
-;; Record indicating the negation existing in the working memory.
-(defrecord NegationResult [gen-rule-name])
+;; A marker interface to identify internal facts.
+(definterface ISystemFact)
```

Comment:
Definterface doesn't exist in ClojureScript at this time [1, 2].  This makes sense since it is fundamentally about a JVM construct that doesn't apply to a JS VM.  This causes the ClojureScript compilation to fail as can be seen in the Travis log for this job.  [3]  Unfortunately this doesn't cause the main build to fail as discussed at https://github.com/cerner/clara-rules/issues/186.

My suggestion would be to make this a protocol instead, which appears to exist in ClojureScript based on usage in cljs.core.  On the JVM that will create a concrete interface that we can use with instance?; in ClojureScript when we address https://github.com/cerner/clara-rules/issues/241 we can use satisfies?.  We could continue to define an interface, derive the type using a custom hierarchy for ClojureScript, and work it out with reader conditionals between Clojure and ClojureScript but I'd prefer to avoid that if we can.

1. http://dev.clojure.org/jira/browse/CLJS-1190
2. https://github.com/clojure/clojurescript/search?utf8=%E2%9C%93&q=definterface
3. https://travis-ci.org/cerner/clara-rules/builds/180306623